phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2102,6 +2106,17 @@ private PTable createTableInternal(CreateTableStatement statement, byte[][] spli
                 }
             }
 
+            Boolean isChangeDetectionEnabledProp =
+                (Boolean) TableProperty.CHANGE_DETECTION_ENABLED.getValue(tableProps);
+            if (isChangeDetectionEnabledProp != null) {
+                if (tableType != TABLE && tableType != VIEW) {
+                    throw new SQLExceptionInfo.Builder(
+                        SQLExceptionCode.CHANGE_DETECTION_SUPPORTED_FOR_TABLES_AND_VIEWS_ONLY)
```

Comment:
If possible, we can refactor this validation in a small util method and it can be used by both workflows: createTable/createIndex and addColumn/alterIndex?

Reply:
Done